cluster/gke/composition.yaml

```
@@ -1,39 +1,96 @@
 apiVersion: apiextensions.crossplane.io/v1
 kind: Composition
 metadata:
-  name: gke.gcp.platformref.crossplane.io
+  name: gke.gcp.platformref.upbound.io
   labels:
     provider: GCP
 spec:
   writeConnectionSecretsToNamespace: upbound-system
   compositeTypeRef:
-    apiVersion: gcp.platformref.crossplane.io/v1alpha1
+    apiVersion: gcp.platformref.upbound.io/v1alpha1
     kind: GKE
   resources:
-    - base:
-        apiVersion: container.gcp.crossplane.io/v1beta2
+    - name: service-account
+      base:
+        apiVersion: cloudplatform.gcp.upbound.io/v1beta1
+        kind: ServiceAccount
+      patches:
+        - fromFieldPath: spec.id
+          toFieldPath: metadata.name
```

```suggestion
          toFieldPath: metadata.annotations[crossplane.io/external-name]
```
This is a better way of naming resources in general so that Crossplane can take care of `metadata.name`, making sure there is no conflict.

---

Ah, got it. I always had an opposite vector of `name -> external-name` in mind. That makes sense, will update